### PR TITLE
Change get_sotkanet default years, add function sotkanet_indicators_metadata and move from httr to httr2

### LINKED ISSUE #12: Default value for years
The default for GetDataSotkanet() could be something else than 1991:2015, as many indicators only have data for latest years.

### LINKED ISSUE #19: Switch from httr to httr2
`httr` package is marked in [R lifecycle terms](https://lifecycle.r-lib.org/articles/stages.html#superseded) as `superseded`. This means generally speaking that

> A softer alternative to deprecation is superseded. A superseded function has a known better alternative, but the function itself is not going away . A superseded function will not emit a warning (since there’s no risk if you keep using it), but the documentation will tell you what we recommend instead .
> 
> Superseded functions will not receive new features, but will receive any critical bug fixes needed to keep it working. In some ways a superseded function is actually safer than a stable function because it’s guaranteed never to change (for better or for worse).

and in the specific case of the `httr` package that

> httr is superseded: only changes necessary to keep it on CRAN will be made.

I think it would make sense to update even existing functions from `httr` to `httr2`, since `httr2` is more modern, is already used in e.g. the `eurostat` package and it is probably easier to maintain functions using `httr2` in the future due to better documentation. Special care has to be taken to ensure that the returned data object is identical to the old one. Error message handling can differ from old implementation. Writing completely new functions in `httr2` is also a no-brainer.

### PR DESCRIPTION
**Pull request by Allaht2**
Change `get_sotkanet` `years` parameter default from `1991:2015` to `NULL`. The default now gives the data from all the available years. Also update the function documentation and add more examples.
Add function `sotkanet_indicators_metadata` and update other functions to use it instead of `SotkanetIndicatorsMetadata`.

### CODE REVIEW

**Comment by Allaht2** [comment]
Move from `httr` to `httr2`. All the old and new functions that used `httr` functions now use `httr2` functions instead.

File: R/sotkanet_indicators.R [hunk 1]
```diff
@@ -75,10 +75,10 @@ sotkanet_indicators <- function(id = NULL, type = "table", lang = "fi", user.age
...
     } else if (type == "table"){
       sotkanet_uri <- paste(sotkanet_uri, id, sep = "/")
-      url_object <- httr::parse_url(sotkanet_url)
+      url_object <- httr2::url_parse(sotkanet_url)
       path <- paste(url_object$path, sotkanet_uri, sep = "")
       url_object$path <- path
-      final_url <- httr::build_url(url_object)
+      final_url <- httr2::url_build(url_object)
 
       res <- sotkanet.json_query(final_url,
```
Comment (pitkant) [inline R/sotkanet_indicators.R:83]: Here flatten = TRUE but in sotkanet.json_query function the parameters is not passed forward anymore (?)

File: R/json.R [hunk 2]
```diff
@@ -1,40 +1,39 @@
...
- 
-  httr_get <- httr::GET(url, httr::user_agent(useragent))
-  httr_content <- httr::content(httr_get, 
-                                as = "text", 
-                                encoding = "UTF-8")
 
-  response <- jsonlite::fromJSON(txt = httr_content, flatten)
+  response <- httr2::request(url) %>% httr2::req_user_agent(useragent) %>%
+    httr2::req_perform() %>% httr2::resp_body_json(simplifyVector = TRUE)
+
```
Comment (pitkant) [inline R/json.R]: Is the output still the same as before? Is simplifyVector essentially the same as flatten = TRUE?

File: R/get_sotkanet.R [hunk 3]
```diff
@@ -22,10 +22,10 @@
 #' @param years vector of years, for example `2015:2018` or `c(2010, 2012, ...)`. Default
 #' value is `NULL`, which gives the data from all the available years.
 #' @param genders vector of genders ('male' | 'female' | 'total'). Some datasets do not
-#' work with the gender value 'total' and return an empty data.frame. In these situations
+#' work with only the gender value 'total' and return an empty data.frame. In these situations
 #' it is advised to check out the [eurostat] package instead.
```
Comment (pitkant) [inline R/get_sotkanet.R]: Maybe remove eurostat reference and instruct user to just try with other options

**Review by pitkant — APPROVED** [review]
Good work

**Comment by pitkant** [comment]
Maybe one final addition: since packages like `kableExtra` and `ggplot2` are only in Suggests and not Depends, their presence/absence could be used to control the evaluation of vignette building? Like this:

```
eval = requireNamespace("somedependency")
eval = !identical(Sys.getenv("SOME_THING_YOU_NEED"), "")
eval = file.exists("credentials-you-need")
```

See [Vignettes - Special considerations for vignette code](https://r-pkgs.org/vignettes.html#sec-vignettes-eval-option) for more info.

File: R/json.R (hunk 2 above)
Comment (Allaht2) [reply R/json.R]: My understanding is that simplifyVector does work in a similar way. At least in my testing the output is similar with simplifyVector.

**Pull request merged**